batch/batch/worker/flock.py

```diff
@@ -7,18 +7,21 @@
 
 
 class Flock:
-    def __init__(self, path, pool=None):
+    def __init__(self, path, pool=None, nonblock=False):
         self.path = os.path.abspath(path)
         self.pool = pool
 
         self.flock_flags = fcntl.LOCK_EX
         self.fds = []
 
+        if nonblock:
+            self.flock_flags |= fcntl.LOCK_NB
```

Comment:
I think this is a hold-over from before my comments to the underlying PR.

Reply:
I need this so that the garbage collector doesn't wait on a file to be downloaded if someone else has a lock on that file. I want it to just give up on that and try locking / deleting the next file.